mh/session.py

```diff
@@ -200,13 +200,14 @@ def leave_city(self):
         DB.leave_city(self)
 
     def try_transfer_city_leadership(self):
-        city = self.get_city()
-        if city.leader == self:
-            if len(city.players) > 1:
-                for player in city.players:
-                    if player != self:
-                        city.leader = player
-                        return player
+        if self.local_info['city_id'] is not None:
+            city = self.get_city()
+            if city.leader == self:
+                if len(city.players) > 1:
```

Comment:
This size check seems unnecessary and is redundant with the `for` loop +`if player != self` check.

Reply:
Fixed in latest commit.